src/Utility/ExprFunctions.C

```diff
@@ -409,27 +409,25 @@ Ret EvalMultiFunction<ParentFunc,Ret>::evaluate (const Vec3& X) const
 }
 
 
-template<>
-Vec3 VecFuncExpr::deriv (const Vec3& X, int dir) const
+template<class ParentFunc, class Ret>
+Ret EvalMultiFunction<ParentFunc,Ret>::deriv (const Vec3& X, int dir) const
 {
-  Vec3 result;
-
-  for (size_t i = 0; i < 3 && i < nsd; ++i)
-    result[i] = p[i]->deriv(X,dir);
+  std::vector<Real> tmp(p.size());
+  for (size_t i = 0; i < p.size(); ++i)
+    tmp[i] = p[i]->deriv(X,dir);
 
-  return result;
+  return Ret(tmp);
```

Comment:
Same concern here is for evaluate method, for the SymmTensor case.

Reply:
same answer.